### PR TITLE
BuildPropertiesPlugin: v0.9.5

### PR DESCRIPTION
**Pull request by mr-archano**
- Added support for android library plugin
- Added `getValue()` to evaluate untyped value of `BuildProperties.Entry`
- Changed `booleanValue()`/`intValue()`/`doubleValue`/`stringValue()` to `getBoolean()`/`getInt()`/`getDouble()`/`getString()`
- Added tests for `BuildProperties` class
- Added tests assessing plugin dependencies
- Added tests assessing generated output of sample app

### CODE REVIEW

**Comment by mr-archano** [comment]
@hal9002 Retest this please.

File: BuildPropertiesPlugin/build.gradle [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ext {
   isLocalDeploy = !project.hasProperty('maven')
   pluginGroup = 'com.novoda.build-properties'
-  pluginVersion = isLocalDeploy ? 'LOCAL' : '0.9.0'
+  pluginVersion = isLocalDeploy ? 'LOCAL' : '0.9.5'
```
Comment (ouchadam) [inline BuildPropertiesPlugin/build.gradle]: didn't fancy using `properties.version`?

File: BuildPropertiesPlugin/plugin/src/main/groovy/com/novoda/buildproperties/BuildPropertiesPlugin.groovy [hunk 2]
```diff
@@ -9,8 +9,10 @@ class BuildPropertiesPlugin implements Plugin<Project> {
     @Override
     void apply(Project project) {
         boolean isAndroidApp = project.plugins.hasPlugin('com.android.application')
-        if (!isAndroidApp) {
-            throw new GradleException("This plugin can be applied only to an Android application project")
+        boolean isAndroidLib = project.plugins.hasPlugin('com.android.library')
+
+        if (!isAndroidApp && !isAndroidLib) {
```
Comment (ouchadam) [inline BuildPropertiesPlugin/plugin/src/main/groovy/com/novoda/buildproperties/BuildPropertiesPlugin.groovy]: worth mentioning that it must be applied after the android plugins or is it obvious?

**Comment by ouchadam** [comment]
looks legit ! 👍

File: BuildPropertiesPlugin/build.gradle (hunk 1 above)
Comment (mr-archano) [inline BuildPropertiesPlugin/build.gradle]: could do both I guess. I like that this way `pluginVersion` is in the root extension, therefore accessible from the buildscript of every subproject

**Comment by ataulm** [comment]
retest this please

File: BuildPropertiesPlugin/plugin/src/main/groovy/com/novoda/buildproperties/BuildPropertiesPlugin.groovy (hunk 2 above)
Comment (mr-archano) [inline BuildPropertiesPlugin/plugin/src/main/groovy/com/novoda/buildproperties/BuildPropertiesPlugin.groovy]: fixed in https://github.com/novoda/spikes/pull/119/commits/224f6a867a40af5380f6b655f8fe07ad16c35338

**Pull request merged**